core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedService.java

```diff
@@ -368,14 +380,68 @@ HttpService withExceptionHandler(HttpService service) {
         }
     }
 
+    static final class FileAggregatedMultipart {
```

Comment:
How about moving up to top class to avoid using `java.nio.file.Path`?
Addtionally, `aggregateMultipart()` could be a static method of `FileAggregatedMultipart`.